.github/actions/check_artifact_exists/main.js

```diff
@@ -0,0 +1,124 @@
+        {
+            owner: owner,
+            repo: repo,
+        },
+        (repoArtifacts) => {
+            // console.log(" ==> repoArtifacts", repoArtifacts);
+            return repoArtifacts.data.filter((a) => {
+                // console.log("==> Artifact check", a);
+                return a.name == name
+            })
+        }
+    );
+
+    console.log("==> maybe goodRepoArtifacts:", goodRepoArtifacts);
+    if (goodRepoArtifacts.length > 0) {
+        return goodRepoArtifacts;
+    }
+
+    // We have not been able to find a repo artifact, it's really no good news
+    return [];
+}
+
+async function main() {
+    try {
+        const token = core.getInput("github_token", { required: true });
+        const [owner, repo] = core.getInput("repo", { required: true }).split("/");
+        const path = core.getInput("path", { required: true });
+        const name = core.getInput("name");
+        const download = core.getInput("download");
+        const client = github.getOctokit(token)
+
+        console.log("==> Repo:", owner + "/" + repo);
+
+        const goodArtifacts = await getGoodArtifacts(client, owner, repo, name);
+        console.log("==> goodArtifacts:", goodArtifacts);
+
+        let artifactStatus = "";
+            if (goodArtifacts.length === 0) {
+            artifactStatus = "missing";
+        } else {
+            artifactStatus = "found";
+        }
+
+        console.log("==> Artifact", name, artifactStatus);
+        console.log("==> download", download);
+
+        core.setOutput("status", artifactStatus);
+
+        if (artifactStatus === "found" && download == "true") {
+            console.log("==> # artifacts:", goodArtifacts.length);
+
+            let artifact = goodArtifacts[0];
+
+            console.log("==> Artifact:", artifact.id)
+
+            const size = filesize(artifact.size_in_bytes, { base: 10 })
+
+            console.log("==> Downloading:", artifact.name + ".zip", `(${size})`)
+
+            const zip = await client.actions.downloadArtifact({
+                owner: owner,
+                repo: repo,
+                artifact_id: artifact.id,
+                archive_format: "zip",
+            })
+
+            const dir = name ? path : pathname.join(path, artifact.name)
+
+            fs.mkdirSync(dir, { recursive: true })
+
+            const adm = new AdmZip(Buffer.from(zip.data))
+
+            adm.getEntries().forEach((entry) => {
+                const action = entry.isDirectory ? "creating" : "inflating"
+                const filepath = pathname.join(dir, entry.entryName)
+                console.log(`  ${action}: ${filepath}`)
+            })
+
+            adm.extractAllTo(dir, true)
+        }
+
+        if (artifactStatus === "missing" && download == "true") {
+            core.setFailed("Required", name, "that is missing");
+        }
+
+        return;
+    } catch (error) {
+        console.error(error);
+        core.setFailed(error.message);
+    }
```

Comment:
What is this adding that simply letting it fail without a global try-catch doesn't do?

Reply:
It might have been me trying to be extra cautious / having proper debug output